src/components/Card.js

```diff
@@ -1,41 +1,74 @@
+import { data } from "autoprefixer";
+
 export default class Card {
-  constructor(data, cardSelector, handleImageClick){
-    this.title = data.title;
-    this.url = data.url;
+  constructor(data, cardSelector, handleImageClick, handleDeleteClick, handleLikeClick, handleDislikeClick){
+    this.name = data.name;
+    this.link = data.link;
+    this._id = data._id;
+    this._isLiked = data.isLiked
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick
+    this._handleLikeClick = handleLikeClick;
+    this._handleDislikeClick = handleDislikeClick;
   };
   
-  _delete(){
-    this._cardElement.remove();
+  getCardId(){
+    return this._id
+  }
+
+  delete(){
+    this._cardElement.remove()
   };
   
   _like(){
-    this._cardElement.querySelector('.card__heart').classList.toggle("card__heart_active");
+    this._cardElement.querySelector('.card__heart').classList.add("card__heart_active");
+    this._handleLikeClick(this.getCardId())
   };
 
+  _dislike(){
+    this._cardElement.querySelector('.card__heart').classList.remove('card__heart_active')
+    this._handleDislikeClick(this.getCardId())
+  }
+
   _setEventListeners(){
     this._cardImageElement = this._cardElement.querySelector('.card__image');
 
     this._cardElement.querySelector('.card__heart').addEventListener('click', ()=>{
-      this._like();
+      if (this._isLiked){
+        this._dislike(this)
+      } else {
+        this._like(this)
+      }
+      this._isLiked = !this._isLiked
     });
 
     this._cardElement.querySelector('.card__delete').addEventListener('click', ()=>{
-      this._delete();
+      this._handleDeleteClick(this)
     });
 
     this._cardImageElement.addEventListener('click', ()=>{
+      // console.log(this.getCardId())
       this._handleImageClick(this);
     });
-    };
+  };
 
-  
+  renderLike(){
+    this._cardElement.querySelector('.card__heart').classList.add('card__heart_active')
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```